clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/services/LaunchTemplateService.java

```diff
@@ -190,12 +190,16 @@ public RequestLaunchTemplateData launchTemplateData(
               .withHttpTokens(description.getRequireIMDV2() ? "required" : ""));
     }
 
-    // instance market options
-    request.withInstanceMarketOptions(
-        new LaunchTemplateInstanceMarketOptionsRequest()
-            .withSpotOptions(
-                new LaunchTemplateSpotMarketOptionsRequest()
-                    .withMaxPrice(description.getSpotPrice())));
+    // instance market options, required when launching spot instances
+    // https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-launchtemplate-launchtemplatedata-instancemarketoptions.html
+    if (description.getSpotPrice() != null) {
```

Comment:
seems worth extracting this block into an `addInstanceMarketOptions(request, price)` method since it is effectively duplicated below

Reply:
Sure, put out a minor refactor PR for this class in https://github.com/spinnaker/clouddriver/pull/5030/files in preparation to address this comment.